### PR TITLE
feat: main header

### PR DESCRIPTION
**Pull request by pdabrowsky**
https://app.asana.com/0/1202705605335915/1202705556727675/f

### CODE REVIEW

**Comment by dominikx96** [comment]
@pdabrowsky why did you force push that?

Screenshot 2022-08-24 at 09 26 19

File: src/assets/Logo/Logo.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import type { LogoProps } from './Logo.types';
```
Comment (dominikx96) [inline src/assets/Logo/Logo.tsx:1]: You can store this logo as .svg file, we don't need that as a component

File: src/assets/Logo/Logo.types.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export type LogoProps = {
+  className: string;
+};
```
Comment (dominikx96) [inline src/assets/Logo/Logo.types.ts:3]: if you change a logo to svg, it's not necessary

File: src/components/MainHeader/MainHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+          <div className="main-header__icons">
+            <Button type="button">
+              <Icon symbol="Loupe" />
+            </Button>
+            <Button type="button">
+              <Icon symbol="Play" />
+            </Button>
+            <Button type="button">
+              <Icon symbol="Person" />
+            </Button>
```
Comment (dominikx96) [inline src/components/MainHeader/MainHeader.tsx:29]: please add a onClick handler with console.log

File: src/components/MainHeader/MainHeader.scss [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    position: absolute;
+    left: 0;
+  }
+}
+
+.main-header__icons {
+  position: absolute;
+  right: 0;
+  display: flex;
+  height: 24px;
```
Comment (dominikx96) [inline src/components/MainHeader/MainHeader.scss:45]: You don't need to hardcode height

File: src/components/MainHeader/MainHeader.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            </a>
+          </li>
+          <li>
+            <a
+              className="main-header__navigation__link"
+              href="/"
+            >
+              EVENTS
+            </a>
+          </li>
```
Comment (dominikx96) [inline src/components/MainHeader/MainHeader.tsx:82]: Create an array with these elements, define types and map that array to a component, you don't need to repeat a code

File: src/components/SocialHeader/SocialHeader.tsx [hunk 6]
```diff
@@ -8,69 +8,69 @@ const SocialHeader = () => {
...
           </a>
           <a
-            className="social-header___link"
+            className="social-header__link"
             href="https://pinterest.com/"
             target="_blank"
             rel="noopener noreferrer"
           >
             <Icon symbol="Printest" />
           </a>
```
Comment (dominikx96) [inline src/components/SocialHeader/SocialHeader.tsx:55]: As above, do not repeat that code

**Review by dominikx96 — CHANGES_REQUESTED** [review]
It doesn't look like as on designs, please check again and ask for review once you test that end to end

File: src/components/MainHeader/MainHeader.scss [hunk 7]
```diff
@@ -0,0 +1,77 @@
+@use '../../styles/variables';
+@use '../../styles/breakpoints';
+
+.main-header {
+  display: flex;
+  flex-direction: column;
+}
+
+.main-header--border {
```
Comment (dominikx96) [inline src/components/MainHeader/MainHeader.scss:9]: Why have you used `--`? It 's BEM modificator and should be used if you e.g. create a new component variant 🤔

File: src/components/MainHeader/MainHeader.tsx [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+const MainHeader = () => {
+  const naviagtion = ['series', 'topics', 'picks', 'special programs', 'nowness awards', 'events'];
+  return (
+    <header className="main-header">
+      <div className="main-header--border">
+        <Container
+          variant="md-2"
+          className="main-header__container"
+        >
+          <Burger className="main-header__burger" />
```
Comment (dominikx96) [inline src/components/MainHeader/MainHeader.tsx:19]: Add the `onClick` callback with `console.log`

File: src/components/MainHeader/MainHeader.tsx [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+              <Icon symbol="Person" />
+            </Button>
+          </div>
+        </Container>
+      </div>
+      <nav className="main-header--border">
+        <ul className="main-header__navigation">
+          {naviagtion.map((el, id) => {
+            return (
+              <li key={id}>
```
Comment (dominikx96) [inline src/components/MainHeader/MainHeader.tsx:47]: Using `index` is not a good practice use as a key, please elaborate why

File: src/components/MainHeader/MainHeader.tsx [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+import { Icon } from '../common/Icon/Icon';
+import { Burger } from '../common/Burger/Burger';
+import { Button } from '../common/Button/Button';
+
+import { ReactComponent as Logo } from '../../assets/img/logo.svg';
+
+import './MainHeader.scss';
+
+const MainHeader = () => {
+  const naviagtion = ['series', 'topics', 'picks', 'special programs', 'nowness awards', 'events'];
```
Comment (dominikx96) [inline src/components/MainHeader/MainHeader.tsx:11]: It should have a structure `id, label, url`

**Review by dominikx96 — CHANGES_REQUESTED** [review]
It should be overflow auto, currently I always have a visible scroll bar:
Screenshot 2022-08-29 at 08 29 25

It looks like, these borders overlap:
Screenshot 2022-08-29 at 08 29 32

File: src/components/MainHeader/MainHeader.tsx (hunk 9 above)
Comment (pdabrowsky) [reply src/components/MainHeader/MainHeader.tsx:47]: Using index can negatively impact performance and may cause issues with component state.

File: src/components/SocialHeader/SocialHeader.tsx [hunk 11]
```diff
@@ -2,81 +2,43 @@ import { Icon } from '../common/Icon/Icon';
...
+  const icons: IconLink[] = [
+    { id: nextIndex++, symbol: 'Facebook', link: 'https://www.facebook.com/' },
+    { id: nextIndex++, symbol: 'Twitter', link: 'https://twitter.com/' },
+    { id: nextIndex++, symbol: 'Instagram', link: 'https://www.instagram.com/' },
+    { id: nextIndex++, symbol: 'Tumblr', link: 'https://www.tumblr.com/' },
+    { id: nextIndex++, symbol: 'Printest', link: 'https://pinterest.com/' },
+    { id: nextIndex++, symbol: 'Youtube', link: 'https://www.youtube.com/' },
+    { id: nextIndex++, symbol: 'Vimeo', link: 'https://vimeo.com/' },
+    { id: nextIndex, symbol: 'Conversation', link: 'https://theconversation.com/' },
+  ];
```
Comment (dominikx96) [inline src/components/SocialHeader/SocialHeader.tsx:20]: Please hardcode ids there is no sense to increment and index here

File: src/components/MainHeader/MainHeader.tsx [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+
+const MainHeader = () => {
+  let nextIndex = 1;
+  const naviagtion: navigation[] = [
+    { id: nextIndex++, label: 'series', url: '/' },
+    { id: nextIndex++, label: 'topics', url: '/' },
+    { id: nextIndex++, label: 'picks', url: '/' },
+    { id: nextIndex++, label: 'special programs', url: '/' },
+    { id: nextIndex++, label: 'nowness awards', url: '/' },
+    { id: nextIndex, label: 'events', url: '/' },
```
Comment (dominikx96) [inline src/components/MainHeader/MainHeader.tsx:20]: same as below

**Review by dominikx96 — CHANGES_REQUESTED** [review]

File: src/components/MainHeader/MainHeader.tsx [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+    { id: 1, label: 'series', url: '/' },
+    { id: 2, label: 'topics', url: '/' },
+    { id: 3, label: 'picks', url: '/' },
+    { id: 4, label: 'special programs', url: '/' },
+    { id: 5, label: 'nowness awards', url: '/' },
+    { id: 6, label: 'events', url: '/' },
+  ];
+  return (
+    <header className="main-header">
+      <div className="main-header-border">
```
Comment (dominikx96) [inline src/components/MainHeader/MainHeader.tsx:23]: Why is it `border`? Also, you should follow BEM here and make sth like:
```
main-header__content
```

File: src/components/MainHeader/MainHeader.tsx [hunk 14]
```diff
@@ -0,0 +1,74 @@
...
+            <Button
+              type="button"
+              onClick={() => console.log('click')}
+            >
+              <Icon symbol="Person" />
+            </Button>
+          </div>
+        </Container>
+      </div>
+      <nav className="main-header-border">
```
Comment (dominikx96) [inline src/components/MainHeader/MainHeader.tsx:55]: Same as above

**Review by dominikx96 — CHANGES_REQUESTED** [review]

**Review by dominikx96 — APPROVED** [review]

**Pull request merged**